### DISCUSSION TITLE
CFY==CSF? And other force questions

### DISCUSSION THREAD

**Discussion by timjim333** [su2code/SU2#1227] [Q&A] [answered]
Hi team,

As always, thanks for the great work you've been doing on SU2. I had some questions on the `forces_breakdown.dat` file and the orientation of axes in the `surface_flow.vtu` file. I've attached an excerpt of one of my breakdown files to facilitate discussion.

1. What produces the difference in the values of CFz and CL? I thought it was purely due to AoA/alignment with the flight path, i.e. CL = CFz*cos(AoA), but this doesn't seem to be the case. In the attached case, if CFz is correct, then CL shouldn't be the prescribed CL=0.11.
2. Is the side force coefficient CSF = CFy (at zero sideslip) and is it calculated in an analogous way to CFz and CFx? I imported the surface flow into Paraview for post-processing and attempted to numerically integrate the pressure to get lift, drag and side force (ultimately, I want to extract bending moments). I did this by extracting the surface normals, then multiplying the normal X, Y, Z with -Pressure, then integrating to get Fx, Fy, and Fz. However, while the Fz and Fx values roughly matched CFz and CFx multiplied by the non-dimensional factor, respectively, Fy was off by an order of magnitude and in the opposite direction. For the direction, I guess it's because the side force is actually opposite to the y-direction, towards the aircraft but I'm not sure why the magnitude it so far off. I'm also not sure why Fx and Fz didn't exactly match, as I assume SU2 is doing the same thing internally! I wasn't sure if this was a bug or if I've not understood correctly.
3. Is the nondimensional factor 0.5*rho*V^2*S_ref? I did a quick check on the values in the breakdown files and it seemed pretty close.

Thanks for your thoughts!
Take care,
Tim

```
Problem definition:

Compressible Euler equations.
Mach number: 2.2.
Angle of attack (AoA): 3.15332 deg, and angle of sideslip (AoS): 0 deg.
Simulation at a cte. CL: 0.11.
Approx. Delta CL / Delta AoA: 0.0146808 (1/deg).
Approx. Delta CD / Delta CL: 0.321123.
Approx. Delta CMx / Delta CL: 0.0815844.
Approx. Delta CMy / Delta CL: -0.172267.
Approx. Delta CMz / Delta CL: 0.0249453.
Dimensional simulation.
The reference area is 268.56 m^2.
The reference length is 52 m.


Problem definition:

Inviscid flow: Computing density based on free-stream
temperature and pressure using the ideal gas law.
Force coefficients computed using free-stream values.
-- Input conditions:
Fluid Model: STANDARD_AIR 
Specific gas constant: 287.058 N.m/kg.K.
Specific gas constant (non-dim): 287.058
Specific Heat Ratio: 1.4000 
Free-stream static pressure: 7171.64 Pa.
Free-stream total pressure: 76684.3 Pa.
Free-stream temperature: 216 K.
Free-stream total temperature: 425.088 K.
Free-stream density: 0.115663 kg/m^3.
Free-stream velocity: (647.203, 0, 35.6531) m/s. Magnitude: 648.184 m/s.
Free-stream total energy per unit mass: 365083 m^2/s^2.
-- Reference values:
Reference specific gas constant: 1 N.m/kg.K.
Reference pressure: 1 Pa.
Reference temperature: 1 K.
Reference density: 1 kg/m^3.
Reference velocity: 1 m/s.
Reference energy per unit mass: 1 m^2/s^2.
-- Resulting non-dimensional state:
Mach number (non-dim): 2.2
Specific gas constant (non-dim): 287.058
Free-stream temperature (non-dim): 216
Free-stream pressure (non-dim): 7171.64
Free-stream density (non-dim): 0.115663
Free-stream velocity (non-dim): (647.203, 0, 35.6554). Magnitude: 648.184
Free-stream total energy per unit mass (non-dim): 365083


Forces breakdown:

Center of Pressure: X=0.161076Z+3.494994 m.

NOTE: Multiply forces by the non-dimensional factor: 6525340.982899, and the reference factor: 1.000000
to obtain the dimensional force.

Total CL:       0.110000 | Pressure (   99%):    0.110000 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CD:       0.023991 | Pressure (   99%):    0.023991 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CSF:      0.025638 | Pressure (   99%):    0.025638 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CL/CD:    4.585001 | Pressure (  100%):    4.585001 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CMx:      0.011809 | Pressure (   99%):    0.011809 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CMy:     -0.007471 | Pressure (  100%):   -0.007471 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CMz:      0.004790 | Pressure (   99%):    0.004790 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CFx:      0.017904 | Pressure (   99%):    0.017904 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CFy:      0.025638 | Pressure (   99%):    0.025638 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
Total CFz:      0.111153 | Pressure (   99%):    0.111153 | Friction (    0%):    0.000000 | Momentum (    0%):    0.000000
```

**Comment by timjim333** [comment, marked as answer]
Ok, 2. was a daft question... I forgot that I was working with a half mesh, so static pressure didn't cancel out. :man_facepalming: My Fy calculation should have been (Pressure - static pressure)*n_y instead. Working backwards to get the CFy, I get a value of around 0.0226 which is less than CFy in the breakdown file. I imagine that the difference in values may be due to the way the normals are calculated - perhaps the difference in the SU2 normal algorithm and ParaView generate normals algorithm differ more widely when attempting to find X and Y normals due to rounding errors, as a large part of a wing is primarily parallel-ish to the X-Y plane. That's my suspicion anyhow.

I've still not figured out the difference in CL and CFz yet though.

**Reply by timjim333** [reply]
Right, Q1 is also my bad (and not a bug). The AoA should also be applied to CFx of course: CL = CFz\*cos(AoA) - CFy\*sin(AoA)

**Reply by pcarruscag** [reply]
In general that is how the reference factor is computed (with values at infinity, which I think are printed to the breakdown file).
For moving grid problems the velocity is derived from MACH_MOTION (a config option) instead.

**Reply by timjim333** [reply]
Thanks, @pcarruscag - do you have any thoughts on why the difference in a manually integrated CFy might be so much less than the SU2 value (0.0226 from Paraview vs 0.025638)? I presume this error carries into the other coefficients too?

**Reply by timjim333** [reply]
On another note, how should one interpret this line in the file? Is it missing a newline or space, perhaps? Thanks.
```
Center of Pressure: X=0.161076Z+3.494994 m.
```

**Reply by pcarruscag** [reply]
How did you compute it in Paraview?

**Reply by pcarruscag** [reply]
As for you other question, it is the equation for a line, x as a function of z, you would choose z along the chordline typically.

**Reply by timjim333** [reply]
> How did you compute it in Paraview?

It's on the way to getting the root bending moment, so I import the surface_flow.vtu, then run the 'Extract Surface' filter, then the 'Generate Surface Normals' filter, then run a calculator filter to calculate the CMx_local array:
```python
CL_local = CP * (n_z * cos(alpha) − n_x * sin(alpha)) / s_ref
CFy_local = CP * n_y / s_ref
CMx_local = (CL_local * (y - y_root) + CFy_local * (z - z_root)) / l_ref

# Which (using SU2 output terms) simplifies to the calculator input string for CMx_local:
f'Pressure_Coefficient * (Normals_Z * cos({alpha}) - Normals_X * sin({alpha})) * (coordsY - {y_root}) - Normals_Y * (coordsZ - {z_root})) / ({l_ref} * {s_ref})'
```
Then I run the 'Integrate Variables' filter to get the magnitude of CMx. I only noticed that the other coefficients were slightly different because I was testing the input string to check the values of CFy etc. Perhaps some of the difference is down to numerical error, unless you can spot another mistake I made in the above definitions!

**Reply by timjim333** [reply]
> As for you other question, it is the equation for a line, x as a function of z, you would choose z along the chordline typically.

Just to clarify, it's on a different axis system to the standard input?

**Reply by pcarruscag** [reply]
No, cartesian coordinates.
CFy should have the largest error (% wise) in a 3 wing problem since it is the smallest projected area and paraview and SU2 do not compute areas the same way (we use the area of the dual elements).
Your CMx calculation does not look right, it should involve CFz and CFy, CL is not perpendicular to the y axis.

**Reply by timjim333** [reply]
Thanks, I suspected that it was to do with the projection of the y-normal.

When you say 'choose z along the chordline', you mean the vertical value perpendicular to the chordline, since the streamwise direction is +x, isn't it?

Regarding CMx, I considered CFz and CFy at first, but then I thought that it might only valid if the flow is at 0 AoA; since CFx,y,z are fixed to the mesh axes, is that correct? If you run the mesh at a non-zero alpha or if you have the geometry meshed in at some angle of attack, then I'm not sure that CFz will give the correct vertical term of the moment. Is it not more correctly associated with the lift component, i.e. CL? I guess in actual fact, one needs to get the force perpendicular to the X-Y plane in the aircraft body axis.

Just to check, are CFx, CFy, and CFz aligned with the mesh and then scaled with AoA to get CL, CD, (and scaled with sideslip to get CSF)? (As an aside, why is CL not perpendicular to the y-axis? Surely all forces in the X-Z plane are perpendicular, be that CL or CD? Perhaps I've just misunderstood you there!) Thanks again, @pcarruscag

**Reply by pcarruscag** [reply]
Formally you intersect the CoP equation with that of the chordline.

I see... Usually people do not rotate the geometry, we just change the direction of the freestream flow, because that is easier.
CL and CD are only aligned with the axis at 0 AoA.
If you are rotating your geometry for some reason you need to use AoA=0 in SU2, and then you need to manually compute CL CD and CSF.

**Reply by timjim333** [reply]
I see, thanks for the clarification on CoP!

Ah, I should have been more specific. I'm rotating my geometries in the mesh as I need to align the shock structures with a structured collar mesh (I'm working on supersonic boom propagation). Currently, I run 2 meshes for a given design - in the first mesh, it's set at zero (or some nominal value) and SU2 is run in fixed CL mode to find the 'correct' AoA increment for that CL, then I re-mesh the geometry at that alpha and run SU2 fixed at 0 AoA (or still in fixed CL mode, assuming that the AoA change will be very small) from which I pull my results. In this case, CL and CD should still be correctly aligned, is that correct?

Can I confirm what you mean by the need to manually compute CD and CL? From my understanding, let's say you set 3 degrees AoA in the mesh and run SU2 at 0 AoA, CL and CD should still be correct, as it's defined as perpendicular and parallel, respectively, to the flight path. On the other hand, if I mesh the geometry at 0 degrees and set SU2 to 3 degrees AoA, then CL and CD should be the same as before, but CFz and CFx will be different. Or so I thought; am I wrong?

More generally, where you say:
> CL and CD are only aligned with the axis at 0 AoA.
this is where CL=CFz and CD=CFx, right? Where some AoA is introduced, I assume SU2 will produce CL =CFz * cos(alpha) − CFx * sin(alpha) and CD = CFz * sin(alpha) + CFx * cos(alpha).

At least, this is what I expect, though I may be wrong. As the 'zero' angle of attack for a given geometry is somewhat arbitrary anyway, so the angle at which it is meshed shouldn't matter as long lift (CL) is perpendicular to the flight path and drag (CD) is parallel (and as long as we use CL and CD instead of CFz and CFy), the result should be the same, so no manual recomputation is required? It would be good to hear your thoughts! Thanks again.

**Reply by pcarruscag** [reply]
You are right I managed to confuse myself. CD and CL are correct as computed, it is xyz you may need to rotate if you need them in some kind of body-aligned coordinates.
But these you rotate by the AoA of the grid and not the AoA of the grid + whatever correction the fixed CL mode gives you.

Mark your original questions as answered if you feel they have been.

**Reply by timjim333** [reply]
Awesome, thanks @pcarruscag I got confused myself, that's why I asked. It's good to have been able to confirm this! I'll mark it as answered until someone else comes along with another theory. Take care.

**Reply by timjim333** [reply]
Ok, sorry,  I can only seem to mark my own original comment as the answer rather than any of the following replies - maybe it's a limitation of the Giyhub Q&A system (or me not knowing how to use it). Thanks for the help!
